src/node/uri_transformer.ts

```diff
@@ -0,0 +1,44 @@
+// NOTE: copied over from lib/vscode/src/vs/common/uriIpc.ts
+// remember to update this for proper type checks!
+
+interface UriParts {
+	scheme: string;
+	authority?: string;
+	path?: string;
+}
+
+interface IRawURITransformer {
+	transformIncoming(uri: UriParts): UriParts;
+	transformOutgoing(uri: UriParts): UriParts;
+	transformOutgoingScheme(scheme: string): string;
+}
+
+// This is deliberate! see src/vs/workbench/services/extensions/node/extensionHostProcessSetup.ts
```

Comment:
What does this comment mean?

I wonder if we should put an explanation of `export =` since the behavior is obscure.

Reply:
Added comments!